js/script.js

```diff
@@ -74,142 +76,128 @@ function addBook(){
 
     const messageElement = document.querySelector('.message');
 
-    if (formId.value !== '' && formTitle.value !== '' && formEditionYear.value !== '' && formAuthor.value !== ''
-        && formCost.value !== '' && formNumberOfCopies.value !== '') {
-        if (books.some(book => book.id_number === newBook.id_number)) {
-            messageElement.textContent = 'The book has already existed.';
-            messageElement.style.color = 'red';
-        } else {
-            books.push(newBook);
-            messageElement.textContent = 'The book is saved.';
-            messageElement.style.color = 'green';
+    if (formEditionYear.value >= 2010 && formEditionYear.value <= 2030 && formCost.value >= 1 && formCost.value <= 1000) {
+        if (formId.value !== '' && formTitle.value !== '' && formEditionYear.value !== '' && formAuthor.value !== ''

```

Comment:
I see this like a defence code which really good , but will user understand why his new book wasn't added to the table in case the value is empty or out of bounds ?